woocommerce/woocommerce-gateway-stripe
# Use original gateway to process payment request buttons from UPE

# Changes proposed in this Pull Request:

When UPE is enabled, the payment request buttons were not working. This was due to the `process_payment` function in the UPE gateway being unable to process the payment request transactions. This issue was [addressed in the WCPay plugin](https://github.com/Automattic/woocommerce-payments/pull/2691) by falling back to the original gateway's `process_payment` function to handle any non-UPE payment methods. I have adopted the same solution here.

This PR updates the UPE gateway to fallback to the original gateway when a payment intent ID was not found during the call the process payment. This approach is essentially saying that any order UPE is unable to properly process (due to missing payment intent ID) should be handled by the original gateway.

# Testing instructions

I  have only been able to test with Google Pay, so I included those instructions here:

- Start with a site with UPE and payment request buttons enabled.
- You will need a Google Pay card set up to test this. You can check this by making sure you are logged in to pay.google.com and have a credit card on file there. To test Google Pay, you will need to be using the Chrome browser.
- Go to a product page and confirm the Google Pay button is displayed.
- Click on Google Pay and go through with the transaction.
- The order should complete successfully with this branch.
- Be sure to do thorough testing of paying with a saved payment method using UPE as well.

-------------------
